### DISCUSSION TITLE
Could not open /sys/fs/cgroup/cgroup.procs

### DISCUSSION THREAD

**Discussion by Zplusless** [srsran/srsRAN_Project#415] [General Help] [answered]
![image](https://github.com/srsran/srsRAN_Project/assets/10062702/5fb30da7-38e8-428e-a00a-cdc82e445334)

The cproup.procs problem starts from commit dcd905c on 20231222. The five recent master commits are tested as follows.

|version               |has this info            |commit date|
|  ----  | ----  | ----  |
|5e6f50a              |    no||
|374200d             |    no||
|55c984b             |    no                       | 20231207|
|dcd905c             |    yes                       | 20231222|
|0b2702c              |   yes                       | 20240115|

The host OS is Ubuntu 20.04 with kernel version 5.4.0. I checked `/sys/fs/cgroup/` and found that there is no cgroup.proc in the directory:
![image](https://github.com/srsran/srsRAN_Project/assets/10062702/82ad5fcb-7fbb-4995-a953-b08da1339e36)

However, `cgroup.proc` could be found in `/sys/fs/cgroup/cpuset`:
![image](https://github.com/srsran/srsRAN_Project/assets/10062702/013350a5-7f18-4e0a-bf40-f2c7117efb03)

I'm not sure if it is ok to create a soft link of `/sys/fs/cgroup/cpuset/cgroup.proc` under `/sys/fs/cgroup/cpuset/` and what the influences of this error message are.

[gnb.log](https://github.com/srsran/srsRAN_Project/files/13987892/gnb.log)

[gnb_zmq-fusion.yaml.txt](https://github.com/srsran/srsRAN_Project/files/13987896/gnb_zmq-fusion.yaml.txt)

**Comment by pgawlowicz** [comment]
Hi @Zplusless, any update on this issue?

**Reply by Zplusless** [reply]
unfortunately not. I'm not sure if it is an OS version problem.

**Reply by pgawlowicz** [reply]
was the gnb running in a VM?

**Reply by Zplusless** [reply]
no, it runs on a server with ubuntu 20.04:
```
CPU(s):                          20
On-line CPU(s) list:             0-19
Thread(s) per core:              2
Core(s) per socket:              10
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       GenuineIntel
CPU family:                      6
Model:                           108
Model name:                      Intel(R) Xeon(R) D-1747NTE CPU @ 2.50GHz
Stepping:                        1
CPU MHz:                         2500.000
CPU max MHz:                     2500.0000
CPU min MHz:                     800.0000
BogoMIPS:                        5000.00
```

**Reply by pgawlowicz** [reply]
do you start only one gnb instance or multiple?

**Reply by Zplusless** [reply]
only one

**Reply by pgawlowicz** [reply]
hmm, I tested your config, and it runs without this issue. Then it might be your machine problem. Could you try with a different PC?

**Reply by Zplusless** [reply, marked as answer]
I tried on another machine(centos stream9 with srsRAN-master-0b2702c), it runs without the cgroup problem. I checked `/sys/fs/cgroup` and find `cgroup.procs`. So I suppose it is a OS problem. Since Ubuntu start to use cgroup v2 by default since ubuntu-21.10(check this [link](https://www.phoronix.com/news/Ubuntu-21.10-systemd-cgroup#:~:text=Ubuntu%2021.10%20Systemd%20To%20Finally%20Ship%20With%20Cgroup%20v2%20By%20Default,-Written%20by%20Michael)). 

Considering cgroup v1 and v2 have different dir structure in `/sys/fs/cgroup`, low version of ubuntu (e.g. ubuntu20.04) may has this output.

Thanks @pgawlowicz, I'll close this discuss as solved.
